src/juju_doctor/builtin/mutually-exclusive-endpoints.py

```diff
@@ -0,0 +1,260 @@
+                            f'"{status_name}".'
+                        )
+
+
+# ==========================
+# Helper functions
+# ==========================
+
+
+def example_status_missing_applications():
+    """Doctest input.
+
+    This deployment status is missing applications.
+    """
+    return {}
+
+
+def example_status_multiple_endpoints_violating():
+    """Invalid topology of grafana-agent and another charm.
+
+    In this status, grafana-agent and foo-charm are inter-related over both of the
+    cos_agent and juju-info interfaces.
+    """
+    return yaml.safe_load("""
+applications:
+  ga-one:
+    charm: grafana-agent
+    relations:
+      cos-agent:
+      - related-application: foo
+        interface: cos_agent
+      juju-info:
+      - related-application: foo
+        interface: juju-info
+  ga-two:
+    charm: grafana-agent
+    relations:
+      cos-agent:
+      - related-application: foo
+        interface: cos_agent
+      juju-info:
+      - related-application: foo
+        interface: juju-info
+  foo:
+    charm: foo-charm
+    relations:
+      foo-cos-agent:
+      - related-application: ga-one
+        interface: cos_agent
+      - related-application: ga-two
+        interface: cos_agent
+      foo-juju-info:
+      - related-application: ga-one
+        interface: juju-info
+      - related-application: ga-two
+        interface: juju-info
+""")
+
+
+def example_status_single_endpoint_violating():
+    """Invalid topology of grafana-agent and another charm.
+
+    In this status, grafana-agent and foo-charm are inter-related over both of the
+    cos_agent and juju-info interfaces.
+    """
+    return yaml.safe_load("""
+applications:
+  ga:
+    charm: grafana-agent
+    relations:
+      cos-agent:
+      - related-application: foo
+        interface: cos_agent
+      juju-info:
+      - related-application: foo
+        interface: juju-info
+  foo:
+    charm: foo-charm
+    relations:
+      foo-cos-agent:
+      - related-application: ga
+        interface: cos_agent
+      foo-juju-info:
+      - related-application: ga
+        interface: juju-info
+""")
+
+
+def example_status_valid():
+    """Valid topology of grafana-agent and other charms.
+
+    In this status, grafana-agent is related to two different charms: foo and bar. For each
+    relation, grafana-agent is related to only one of the cos_agent and juju-info interfaces.
+    """
+    return yaml.safe_load("""
+applications:
+  ga:
+    charm: grafana-agent
+    relations:
+      cos-agent:
+      - related-application: foo
+        interface: cos_agent
+      juju-info:
+      - related-application: bar
+        interface: juju-info
+  foo:
+    charm: foo-charm
+    relations:
+      foo-cos-agent:
+      - related-application: ga
+        interface: cos_agent
+  bar:
+    charm: bar-charm
+    relations:
+      bar-juju-info:
+      - related-application: ga
+        interface: juju-info
+""")
+
+
+def example_args_without_ignore():
```

Comment:
All these args functions are so short it may be clearer to inline them in the doctests above. Up to you.